crypto/aes/asm/aes-riscv64-zvkned.pl

```diff
@@ -0,0 +1,1413 @@
+    addi $KEYP, $KEYP, 16
+    @{[vle32_v $v10, $KEYP]}
+    addi $KEYP, $KEYP, 16
+    @{[vle32_v $v11, $KEYP]}
+___
+
+    return $code;
+}
+
+# Load all 13 round keys to v1-v13 registers.
+sub aes_192_load_key {
+    my $code=<<___;
+    @{[vsetivli "zero", 4, "e32", "m1", "ta", "ma"]}
+    @{[vle32_v $v1, $KEYP]}
+    addi $KEYP, $KEYP, 16
+    @{[vle32_v $v2, $KEYP]}
+    addi $KEYP, $KEYP, 16
+    @{[vle32_v $v3, $KEYP]}
+    addi $KEYP, $KEYP, 16
+    @{[vle32_v $v4, $KEYP]}
+    addi $KEYP, $KEYP, 16
+    @{[vle32_v $v5, $KEYP]}
+    addi $KEYP, $KEYP, 16
+    @{[vle32_v $v6, $KEYP]}
+    addi $KEYP, $KEYP, 16
+    @{[vle32_v $v7, $KEYP]}
+    addi $KEYP, $KEYP, 16
+    @{[vle32_v $v8, $KEYP]}
+    addi $KEYP, $KEYP, 16
+    @{[vle32_v $v9, $KEYP]}
+    addi $KEYP, $KEYP, 16
+    @{[vle32_v $v10, $KEYP]}
+    addi $KEYP, $KEYP, 16
+    @{[vle32_v $v11, $KEYP]}
+    addi $KEYP, $KEYP, 16
+    @{[vle32_v $v12, $KEYP]}
+    addi $KEYP, $KEYP, 16
+    @{[vle32_v $v13, $KEYP]}
+___
+
+    return $code;
+}
+
+# Load all 15 round keys to v1-v15 registers.
+sub aes_256_load_key {
+    my $code=<<___;
+    @{[vsetivli "zero", 4, "e32", "m1", "ta", "ma"]}
+    @{[vle32_v $v1, $KEYP]}
+    addi $KEYP, $KEYP, 16
+    @{[vle32_v $v2, $KEYP]}
+    addi $KEYP, $KEYP, 16
+    @{[vle32_v $v3, $KEYP]}
+    addi $KEYP, $KEYP, 16
+    @{[vle32_v $v4, $KEYP]}
+    addi $KEYP, $KEYP, 16
+    @{[vle32_v $v5, $KEYP]}
+    addi $KEYP, $KEYP, 16
+    @{[vle32_v $v6, $KEYP]}
+    addi $KEYP, $KEYP, 16
+    @{[vle32_v $v7, $KEYP]}
+    addi $KEYP, $KEYP, 16
+    @{[vle32_v $v8, $KEYP]}
+    addi $KEYP, $KEYP, 16
+    @{[vle32_v $v9, $KEYP]}
+    addi $KEYP, $KEYP, 16
+    @{[vle32_v $v10, $KEYP]}
+    addi $KEYP, $KEYP, 16
+    @{[vle32_v $v11, $KEYP]}
+    addi $KEYP, $KEYP, 16
+    @{[vle32_v $v12, $KEYP]}
+    addi $KEYP, $KEYP, 16
+    @{[vle32_v $v13, $KEYP]}
+    addi $KEYP, $KEYP, 16
+    @{[vle32_v $v14, $KEYP]}
+    addi $KEYP, $KEYP, 16
+    @{[vle32_v $v15, $KEYP]}
+___
+
+    return $code;
+}
+
+# aes-128 encryption with round keys v1-v11
+sub aes_128_encrypt {
+    my $code=<<___;
+    @{[vaesz_vs $v24, $v1]}     # with round key w[ 0, 3]
+    @{[vaesem_vs $v24, $v2]}    # with round key w[ 4, 7]
+    @{[vaesem_vs $v24, $v3]}    # with round key w[ 8,11]
+    @{[vaesem_vs $v24, $v4]}    # with round key w[12,15]
+    @{[vaesem_vs $v24, $v5]}    # with round key w[16,19]
+    @{[vaesem_vs $v24, $v6]}    # with round key w[20,23]
+    @{[vaesem_vs $v24, $v7]}    # with round key w[24,27]
+    @{[vaesem_vs $v24, $v8]}    # with round key w[28,31]
+    @{[vaesem_vs $v24, $v9]}    # with round key w[32,35]
+    @{[vaesem_vs $v24, $v10]}   # with round key w[36,39]
+    @{[vaesef_vs $v24, $v11]}   # with round key w[40,43]
+___
+
+    return $code;
+}
+
+# aes-128 decryption with round keys v1-v11
+sub aes_128_decrypt {
+    my $code=<<___;
+    @{[vaesz_vs $v24, $v11]}   # with round key w[43,47]
+    @{[vaesdm_vs $v24, $v10]}  # with round key w[36,39]
```

Comment:
In the comment Is the difference between `v11` indices between encryption and decruption expected ?
```suggestion
    @{[vaesz_vs $v24, $v11]}   # with round key w[40,43]
    @{[vaesdm_vs $v24, $v10]}  # with round key w[36,39]
```